### PR TITLE
[RUN-4602] Persist useName in job reference step to prevent UUID reversion

### PR DESCRIPTION
**Pull request by jayas006**
## Release Notes

Fixed an issue in the workflow editor where a Job Reference step set to reference a job by name would revert to referencing by UUID the next time the step was edited. The name-vs-UUID selection is now saved with the step and preserved across edits.

## Problem

When a user creates a Job Reference step using UUID and then edits it to use Name instead, the next edit reverts back to UUID. The selected reference mode was never persisted in the workflow JSON.

## Root Cause

`JobRefFormFields.vue` stored the name/UUID toggle in a local `isUseName` data property. This value was never written back to `modelValue.jobref.useName`, so the parent `JobRefForm` emitted `update:modelValue` without `useName` in the payload. On the next mount, the form had no `useName` to read and defaulted to UUID mode.

## Fix

- **`JobRefFormFields.vue`**: Replaced the local `isUseName` data prop with a computed getter/setter that reads from `modelValue.useName` and emits `update:modelValue` on change.
- **`JobRefForm.vue`**: Added `inferUseName()` called in `mounted()` and the `modelValue` watcher to backfill `useName` for existing job refs saved before this fix (`name` present and no UUID → name mode; UUID present → UUID mode).
- **`types/workflowTypes.ts`**: Added `useName?: boolean` to `JobRefDefinition`.

## Tests

- Switching to name mode → saved payload contains `useName: true`
- Staying in UUID mode → saved payload contains `useName: false`
- Re-opening a form previously saved with `useName: true` → name fields are editable (regression guard for the exact customer scenario)
- Backward compat: existing refs with name and no UUID infer name mode; refs with UUID infer UUID mode

### CODE REVIEW

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue [hunk 1]
```diff
@@ -115,13 +111,26 @@ export default defineComponent({
...
+    this.editModel = merge(this.editModel, this.modelValue);
+    this.inferUseName();
+  },
   methods: {
+    inferUseName() {
+      const ref = this.editModel.jobref;
+      if (ref && ref.useName === undefined) {
+        ref.useName = Boolean(ref.name && !ref.uuid);
+      }
+    },
```
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue]: Fixed this

**Review by smartinellibenedetti — CHANGES_REQUESTED** [review]
So the core fix is right: moving the toggle into a `useName` computed getter/setter in `JobRefFormFields.vue` (backed by `modelValue.useName`) is the correct, shared fix. Both editor entry points render `JobRefFormFields` (the `JobRefForm` modal and the `EditStepCard` card view), so that single change covers every path. 👍

Two things need to change.

**1. Remove `inferUseName()` in `JobRefForm` — it's not possible to reliably infer what was the original selection.**

Because the saved jobref object will always contain both the name and the uuid.

**2. The `useName: false` default is missing from all three jobref default objects.**

There are three hand-copied jobref defaults, each commented *"matching JobRefForm"*, each merged over incoming `modelValue`. None of them currently sets `useName`, so `useName: false` is never persisted explicitly for a never-toggled UUID ref (it only resolves to `false` implicitly via the getter's `?? false`). To make the fix consistent and satisfy the "stays in UUID mode → `useName: false`" expectation across both editors:

| # | File | Location | Repo |
|---|------|----------|------|
| 1 | `JobRefForm.vue` | `editModel.jobref` default | rundeck (this PR) |
| 2 | `EditStepCard.vue` | `jobRefDefaults` (~line 199) | rundeck (this PR) |
| 3 | `EditStepCard.vue` | `jobRefDefaults` (~line 337) | **rundeckpro** (separate repo — follow-up PR) |

`#1` and `#2` belong in this PR. `#3` lives in the rundeckpro repo and must be handled in a separate rundeckpro PR — flagging it so it isn't missed, since the card editor there shares the same `JobRefFormFields` and will otherwise drift.

**Suggested improvement (optional but recommended):** these three defaults are duplicated verbatim and have already drifted once. Consider extracting a single `createJobRefDefaults()` factory (alongside `stepEditorUtils`) as the single source of truth, and have `JobRefForm` and both `EditStepCard`s consume it. That removes the class of bug where a jobref field is added in one default and forgotten in the others — which is precisely what happened here with `useName`.

Types are already correct (`JobRefDefinition` and `JobRefModel` both declare `useName?: boolean`); no other type changes are needed.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue [hunk 2]
```diff
@@ -117,11 +93,16 @@ export default defineComponent({
       this.editModel = merge(this.editModel, val);
     },
```
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue:94]: Hi @smartinellibenedetti ,I applied your feedback (removed inferUseName(), added useName: false in the shared defaults).
Copilot also commented about merge/stale useName, but I agree we shouldn’t infer in JobRefForm since chooser refs usually have both name and uuid.
One edge case: very old refs with name only, no uuid. Backend still runs them by name. In the editor they might show as UUID until re-saved. Is that OK for this PR?
([JobExec.groovy L322](https://github.com/rundeck/rundeck/blob/main/rundeckapp/grails-app/domain/rundeck/JobExec.groovy#L322)).

[the remark this answers is not in the document]
Comment (smartinellibenedetti) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue:94]: That edge case is very unlikely to happen, it would only happen if a job was created before the useName field was introduced (2019), and never updated since then. 

Don't need to address it, but the appropriate place to add a safeguard would be in the computed get for isUseName:
```this.modelValue.useName ?? (!!this.modelValue.name && !this.modelValue.uuid) ```

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefFormFields.vue [hunk 3]
```diff
@@ -73,9 +75,7 @@
             :title="$t('select.an.existing.job.to.use')"
             @click="openJobBrowser"
           >
-            {{
-              $t(!jobBrowserLoading ? "choose.a.job..." : "loading.text")
-            }}
+            {{ $t(!jobBrowserLoading ? "choose.a.job..." : "loading.text") }}
           </span>
         </div>
         <div class="col-sm-10 col-sm-offset-2" style="margin-top: 1em">
```
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefFormFields.vue:81]: Out of scope for this PR — pre-existing style, not part of my changes.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue [hunk 4]
```diff
@@ -115,13 +111,31 @@ export default defineComponent({
...
       this.editModel = merge(this.editModel, val);
+      this.inferUseName(val);
     },
   },
+  mounted() {
+    this.editModel = merge(this.editModel, this.modelValue);
+    this.inferUseName(this.modelValue);
+  },
   methods: {
+    inferUseName(source) {
```
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue]: No code change needed; that method is gone.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/JobRefForm.vue:94]: not adding infer/rebuild in JobRefForm for this PR. The edge case is acceptable to skip; optional safeguard would live in JobRefFormFields if we ever need it.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/tests/JobRefForm.spec.ts [hunk 5]
```diff
@@ -110,6 +141,63 @@ describe("JobRefForm", () => {
...
+        jobref: { ...baseJobRef, uuid: "abc-123" },
+      });
+
+      expect(
+        wrapper.find("[data-testid='jobNameField']").attributes("readonly"),
+      ).toBeDefined();
+      expect(
+        wrapper.find("[data-testid='jobGroupField']").attributes("readonly"),
+      ).toBeDefined();
+    });
```
[the remark this answers is not in the document]
Comment (jayas006) [reply rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/workflow/tests/JobRefForm.spec.ts:198]: Not adding this test  — inference was removed per review; that backward-compat case is out of scope for this PR.

**Review by smartinellibenedetti — APPROVED** [review]
LGTM and thank you for also adding the logic in the stepEditorUtils!
